### ISSUE TITLE
`Lint/UnusedMethodArgument` false positive

### ISSUE THREAD

**Issue by coaxial** [rubocop/rubocop#11129]
`Lint/UnusedMethodArgument` triggers when the argument is actually used in the method.

--------

## Expected behavior

Don't trigger `Lint/UnusedMethodArgument`

## Actual behavior

`Lint/UnusedMethodArgument` was triggered.

```
Inspecting 1 file
W

Offenses:

foo.rb:1:1: C: Style/Documentation: Missing top-level documentation comment for class MyJob.
class MyJob < ApplicationJob
^^^^^^^^^^^
foo.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
class MyJob < ApplicationJob
^
foo.rb:4:24: W: [Correctable] Lint/UnusedMethodArgument: Unused method argument - path_for. If it's necessary, use _ or _path_for as an argument name to indicate that it won't be used.
  def perform(mymodel, path_for = ActiveStorage::Blob.service.path_for)
                       ^^^^^^^^

1 file inspected, 3 offenses detected, 2 offenses autocorrectable
```

## Steps to reproduce the problem

```ruby
class MyJob < ApplicationJob
  queue_as :default

  def perform(mymodel, path_for = ActiveStorage::Blob.service.path_for)
    path = path_for(mymodel.attachment.key)
    do_the_thing(path)
  end
end

```

Then run `rubocop myfile.rb` and see `Lint/UnusedMethodArgument` being triggered.

## RuboCop version

```
$ rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2) [x86_64-linux]
  - rubocop-rails 2.16.1
  - rubocop-rspec 2.13.2
```

**Comment by coaxial** [comment]
Ok, nevermind, I can't Ruby...
